### DISCUSSION TITLE
Openboard on Wayland

### DISCUSSION THREAD

**Discussion by qqqqqqqqq9** [OpenBoard-org/OpenBoard#1178] [General]
I want to summarize the current situation here to help plotting a way forward.

OpenBoard on Xorg is the most mature combination.
However, using Xorg in a sensitive environment such as schools is not advisable.

I found Openboard on Wayland mostly usable but crashes do occur. 

There are some restrictions either to the security concept of wayland or known bugs.
1. Screenshots: applications are required to use xdg-desktop-portal to access the content of other windows. 
   This is not implemented (yet).
2. Desktop mode: Openboard needs to stay on top of all other windows. You must configure your compositor appropriately. 
3. There known bugs. 
      a) invisible main window: https://github.com/OpenBoard-org/OpenBoard/issues/1164 
      b) a crash when configuring multi-screen-setups 
         https://github.com/OpenBoard-org/OpenBoard/issues/843

I'm use kwin-wayland with focus-follows-mouse on window-raise-on-click disabled.
This works fairly well.
If you have other setups that work (especially on gnome), please share.

**Comment by letsfindaway** [comment]
I now created PR #1299 addressing the points listed above:

1. Screenshots in Desktop mode use the xdg-desktop-portal, also screen mirroring and podcast recording using the ScreenCast portal function.
2. To stay on top of other windows in Desktop mode on KDE, I added a KWin script which sets `keepAbove=true`. I found out that on Gnome this not necessary.
3. The "invisible main window" can no longer be reproduced. The crash configuring multi-screen setups was fixed.
